Sprint 04

DanteIT94 opened the pull request:
In accordance with specification of Yandex Practicum (Sprint № 4):
1) Launch Screen (identical to the layout in Figma);
2) The main screen of the application, identical to the layouts in Figma, with the implemented logic.

DanteIT94 committed:
In accordance with specification of Yandex Practicum (Sprint № 4):
1) Launch Screen (identical to the layout in Figma);
2) The main screen of the application, identical to the layouts in Figma, with the implemented logic.

DanteIT94 committed:
Minor fix

DanteIT94 committed:
Sprint_04

Realising technical task for Sprint № 4:
1) Application launch screen (Launch Screen), identical to the layout in Figma.
2) The main screen of the application, identical to the layouts in Figma, with the implemented logic.

IBDesignable commented on MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -1,4 +1,9 @@
 
-extension UIColor { }
+extension UIColor {
+    static let YPGreen = UIColor(red: 96/255, green: 194/255, blue: 142/255, alpha: 1)
+    static let YPRed = UIColor(red: 245/255, green: 107/255, blue: 108/255, alpha: 1)
+    static let YPWhite = UIColor(red: 255/255, green: 255/255, blue: 255/255, alpha: 1)
+    static let YPBlack = UIColor(red: 26/255, green: 27/255, blue: 34/255, alpha: 1)
```
Лучше использовать цвета из ассетов (они же туда зачем-то добавлены), инициализатор с парамтером named

DanteIT94 replied:
Сделано, спасибо за подсказку)

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
 final class MovieQuizViewController: UIViewController {
     // MARK: - Lifecycle
+
+    @IBOutlet weak var yesButton: UIButton!
+    @IBOutlet weak var noButton: UIButton!
```
private

DanteIT94 replied:
Исправлено!

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
+    QuizQuestion(image: "The Green Knight", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: true),
+    QuizQuestion(image: "Old", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false),
+    QuizQuestion(image: "The Ice Age Adventures of Buck Wild", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false),
+    QuizQuestion(image: "Tesla", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false),
+    QuizQuestion(image: "Vivarium", text: "Рейтинг этого фильма больше чем 6?", correctAnswer: false)
+    ]
     override func viewDidLoad() {
         super.viewDidLoad()
+        let viewModel = convert(model: currentQuestion)
+        show(quiz: viewModel)
+        yesButton.transform = CGAffineTransform(scaleX: 0.9, y: 0.9)
```
давай поправим форматриование: cmd+a и control+i 

Также, договоримся, что между логическими блоками будем оставлять одну пустую строчку

DanteIT94 replied:
Прошу Вас уточнить, а что такое "Логический блок" в коде?

IBDesignable replied:
Денис, под логическим блоком я имею ввиду функции, классы и структуры, расширения классов. Также ты можешь написать одну очень большую функцию, но явно отделить шаги алгоритма пустыми строками. 

Это все не влияет на решение задачи в спринте, однако аккуратное ведение кода очень пригодится на работе 👍🏻

спасибо что уточнил, буду четче формулировать это выражение)

DanteIT94 replied:
Понял, принял! Огромное спасибо за разъяснение!)

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
+        let alert = UIAlertController( title: result.title,
+                                       message: result.text,
+                                       preferredStyle: .alert)
+        let action = UIAlertAction(title: result.buttonText, style: .default, handler: { _ in self.currentQuestionIndex = 0
```
здесь принято начинать с новой строки
```suggestion
        let action = UIAlertAction(title: result.buttonText, style: .default, handler: { _ in 
            self.currentQuestionIndex = 0
```

DanteIT94 replied:
Исправлено!

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
+        imageLabel.layer.masksToBounds = true // даём разрешение на рисование рамки
+        imageLabel.layer.borderWidth = 8 // толщина рамки
+        imageLabel.layer.borderColor = isCorrect ? UIColor.YPGreen.cgColor : UIColor.YPRed.cgColor
+        imageLabel.layer.cornerRadius = 6 // радиус скругления углов рамки
```
не по макету

Это свойство можно настроить один раз: на сториборде или во viewDidLoad

DanteIT94 replied:
Исправил на imageLabel.layer.cornerRadius = 20 (как в макете)

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
+        if isCorrect {
+            correctAnswers += 1
+        }
+        imageLabel.layer.masksToBounds = true // даём разрешение на рисование рамки
```
аналогично, это свойство достаточно настроить один раз

DanteIT94 replied:
Принято

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.showNextQuestionOrResult()
+        }
+        
+        }
+    private func showNextQuestionOrResult() {
```
давай здесь и далее добавим отбивку одной пустой строкой

DanteIT94 replied:
Не понимаю, что это значит)

IBDesignable replied:
Здесь как в предыдущем комменатрии конец одной функции и начало другой слипаются. Обычно принято оставлять немного пустого пространство, чтобы при беглом чтении кода можно было быстро соориентироваться

DanteIT94 replied:
Понял)

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
+            imageLabel.layer.borderWidth = 0
+        }
+    }
+    @objc func yesButtonPressed() {
```
private

DanteIT94 replied:
Сделано

IBDesignable commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,15 +2,164 @@ import UIKit
- Mock-данные
- 
+   
+
+
+/*struct ViewModel {
+  let image: UIImage
+  let question: String
+  let questionNumber: String
+}*/
+
```
давай договоримся, что на ревью закомменченный код не будем отправлять. Это мертвый код, который не нужен :)

DanteIT94 replied:
Огромное спасибо! Возможно, пропустил когда делал финальный чек-ап)

DanteIT94 committed:
Спринт № 4 - работа на ошибками Ревью

1) Critical comments were corrected in accordance with the tips of the reviewer.
2) The function of blocking buttons after pressing was added

DanteIT94 committed:
Minor fixs (Launch screen and CornerRadius)

Minor fixs (Sprint 04)
